cirq-core/cirq/protocols/kraus_protocol_test.py

```diff
@@ -171,3 +171,43 @@ def test_has_kraus_when_decomposed(decomposed_cls) -> None:
     op = HasKrausWhenDecomposed(decomposed_cls).on(cirq.NamedQubit('test'))
     assert cirq.has_kraus(op)
     assert not cirq.has_kraus(op, allow_decompose=False)
+
+
+def test_kraus_fallback_to_apply_channel() -> None:
+    """Kraus protocol falls back to _apply_channel_ when no _kraus_, _mixture_, or _unitary_."""
+    p = 0.5
+    K0 = np.sqrt(1 - p) * np.eye(2)
+    K1 = np.sqrt(p) * np.array([[0, 1], [1, 0]])
+    expected_kraus = (K0, K1)
```

Comment:
Could you add some more complex tests? I'm worried there could be lots of ways bit flip with p=0.5 could succeed even if the implementation was wrong. Also, instead of defining expected_kraus from scratch here, prefer using `cirq.kraus(cirq.BitFlipChannel)` (i.e. use the real BitFlipChannel) to ensure the expected values are correct. (It's been a long time; I can't remember offhand how Kraus operators are defined).

One good test could maybe use ResetGate since it has both protocols defined. Create instances that undefine one or the other of those protocol methods and then check that their superoperators are the same.